morphir/src/org/finos/morphir/runtime/ErrorUtils.scala

```diff
@@ -0,0 +1,107 @@
+package org.finos.morphir.runtime
+
+import fansi.Str
+import org.finos.morphir.datamodel.{Concept, Data}
+import org.finos.morphir.naming.*
+import org.finos.morphir.ir.{Type as T, Value as V}
+import org.finos.morphir.ir.Value.{Pattern, TypedValue, Value, USpecification as UValueSpec}
+import org.finos.morphir.ir.Type.{Type, UType, USpecification as UTypeSpec}
+import org.finos.morphir.ir.printing.PrintIR
+
+object ErrorUtils {
+  implicit class ErrorInterpolator(sc: StringContext) {
+    //format: off
+    /**
+     * This interpolator is intended to format errors with potentially bulky arguments which we wish to display as both
+     * IR and native elm. As such it takes the IR arguments and replaces them with a numeric ID in the string, and then
+     * prints them after the message in formatted blocks. For instance:
+     *
+     * Value type does not match declared type: (0) vs (1)
+     * \========== 0: ==========
+     * \| ====Elm====
+     * \| | (1, 2)
+     * \| ====IR====
+     * \| | T.Tuple(T.Ref(int), T.Ref(int))
+     * \========== 1: ==========
+     * \| ====Elm====
+     * \| | ()
+     * \| ====IR====
+     * \| | T.Unit(())
+     *
+     * @param args
+     *   Either IR or non-IR elements to be displayed; IR elements will be handled specially.
+     * @return
+     *   A human-readable error with the IR elements formatted for display after the error message.
+     */
+      //format: on
+    def err(args: Any*): String = {
+      val indexed = args.zipWithIndex
+      val irArgs = indexed
+        .filter { case (arg, _) => isIR(arg) }
+        .zipWithIndex
+        .map { case ((arg, mainIndex), argIndex) => (mainIndex, (argIndex, arg)) }
+        .toMap
+      val processed = indexed.map { case (arg, mainIndex) =>
+        irArgs.get(mainIndex) match {
+          case Some((irIndex, _)) => s"($irIndex)"
+          case None               => arg.toString
+        }
+      }
+      val explanation = sc.s(processed: _*)
+      val terms = irArgs.values.map { case (argIndex, arg) =>
+        process(s" $argIndex: ", arg)
+      }.mkString
+      explanation + "\n" + terms
+    }
+
+    def process(title: String, astLike: Any): String = {
+      val elmBody = barIndentBlock("Elm", astLike.toString)
+      val irBody  = barIndentBlock("IR", PrintIR(astLike).plainText)
```

Comment:
I don't like this `.plainText` call as it loses all of the colors that came out of PrintIR. However, fansi.Str doesn't have a "split" method which I need for the indentation formatting, and just using `.toString` causes the colorations to leak for the rest of the console output. @deusaquilus lmk if you have any ideas?